sharepoint/sharepoint-ps/sharepoint-online/Get-SPOCopilotAgentInsightsReport.md

```diff
@@ -0,0 +1,165 @@
+| Template                | The Site template of the SharePoint site.                |
+| Site owner              | Name of the owner of the SharePoint site.                |
+| Copilot agents          | Number of Copilot agents on the SharePoint site.         |
+| Sensitivity             | The sensitivity label of the SharePoint site.            |
+| Restrict site access    | RAC status (Yes/No) of the SharePoint site.              |
+| Restrict site discovery | RCD status (Yes/No) of the SharePoint site.              |
+| External sharing        | External Sharing status (Yes/No) of the SharePoint site. |
+
+If this cmdlet is executed with `-ReportId` as parameter and `-Content` as `SiteDistribution`, a report showing Copilot distribution across sites in last N days will be displayed with the following properties:
+
+| Property       | Description                                                                                  |
+|:---------------|:---------------------------------------------------------------------------------------------|
+| Site template  | The Site template of the SharePoint site.                                                    |
+| Sites          | Number of sites corresponding to that particular site template.                              |
+| Copilot agents | Name of Copilot agent on the SharePoint site corresponding to that particular site template. |
+
+If this cmdlet is executed with both the parameters, i.e. `-ReportId` and `-Action`, and if the value of `-Action` is set as `View`, it will display the same result as described above. If the value of `-Action` is set to `Download`, it will download the full report in CSV format to the same path from where the command was run.
+  
+> [!NOTE]
+> All reports adhere to any retention timeline as per [Data Access Governance](/sharepoint/data-access-governance-reports).
+
+## EXAMPLES
+
+### -----------------------EXAMPLE 1-----------------------------
+
+```powershell
+Get-SPOCopilotAgentInsightsReport
+```
+
+Example 1 enables administrator to view the status of all active and completed reports.
+
+### -----------------------EXAMPLE 2-----------------------------
+
+```powershell
+Get-SPOCopilotAgentInsightsReport –ReportId 9d946216-afe7-49f5-8267-7b662435c70b
+```
+
+Example 2 enables administrator to view the Copilot agent insights report of ReportId: `9d946216-afe7-49f5-8267-7b662435c70b`.
+
+### -----------------------EXAMPLE 3-----------------------------
+
+```powershell
+Get-SPOCopilotAgentInsightsReport – ReportId 9d946216-afe7-49f5-8267-7b662435c70b -Action Download
+```
+
+Example 3 enables administrator to download the Copilot agent insights report of ReportId: `9d946216-afe7-49f5-8267-7b662435c70b` to the same path from where the command was run.
+
+## PARAMETERS
+
+### -ReportId
+
+It is an optional parameter, and it specifies the unique Id of the report to be viewed or downloaded.
```

Comment:
>It is an optional parameter,  [](http://example.com/codeflow?start=0&length=29)

see other PR comment, here and elsewhere

Reply:
updated the description of the parameter by removing the `optional` keyword. Marking as resolved.